routers/api/v1/repo/issue.go

```diff
@@ -304,25 +322,33 @@ func ListIssues(ctx *context.APIContext) {
 	// Only fetch the issues if we either don't have a keyword or the search returned issues
 	// This would otherwise return all issues if no issues were found by the search.
 	if len(keyword) == 0 || len(issueIDs) > 0 || len(labelIDs) > 0 {
-		issues, err = models.Issues(&models.IssuesOptions{
+		issuesOpt := &models.IssuesOptions{
 			ListOptions:  listOptions,
 			RepoIDs:      []int64{ctx.Repo.Repository.ID},
 			IsClosed:     isClosed,
 			IssueIDs:     issueIDs,
 			LabelIDs:     labelIDs,
 			MilestoneIDs: mileIDs,
 			IsPull:       isPull,
-		})
-	}
+		}
 
-	if err != nil {
-		ctx.Error(http.StatusInternalServerError, "Issues", err)
-		return
-	}
+		if issues, err = models.Issues(issuesOpt); err != nil {
+			ctx.Error(http.StatusInternalServerError, "Issues", err)
+			return
+		}
 
-	ctx.SetLinkHeader(ctx.Repo.Repository.NumIssues, listOptions.PageSize)
-	ctx.Header().Set("X-Total-Count", fmt.Sprintf("%d", ctx.Repo.Repository.NumIssues))
+		issuesOpt.ListOptions = models.ListOptions{
```

Comment:
Why is this needed?

Reply:
The maximum value that matches the filter is needed to calculate the last page number.
So, It need a value from the zero to tail.

Reply:
There coud be a if statement in the setupSession for IssueListOptions when page==0 ignore limint but you have to look into the code